apps/server-asset-sg/src/features/assets/workflow/workflow.service.ts

```diff
@@ -17,6 +30,26 @@ export class WorkflowService {
         HttpStatus.UNPROCESSABLE_ENTITY,
       );
     }
+    if (change.assigneeId) {
+      const newAssignee = await this.userRepo.find(change.assigneeId);
+      if (!newAssignee) {
+        throw new HttpException(`Assignee with ID ${change.assigneeId} not found.`, HttpStatus.UNPROCESSABLE_ENTITY);
+      }
+      const assigneeRole = newAssignee.roles.get(workflow.workgroupId);
+      if (!assigneeRole) {
+        throw new HttpException(
+          `Assignee with ID ${change.assigneeId} does not have a role in workgroup ${workflow.workgroupId}.`,
+          HttpStatus.UNPROCESSABLE_ENTITY,
+        );
+      }
+      if (getRoleIndex(assigneeRole) < getRoleIndex(getRoleForStatus(change.status))) {
+        throw new HttpException(
+          "The selected assignee's role is not sufficient for the new status.",
+          HttpStatus.UNPROCESSABLE_ENTITY,
+        );
+      }
```

Comment:
You could shorten this and also hide implementation details by simply using the policy.
```suggestion
const policy = new WorkflowPolicy(newAssignee);
if (!policy.canUpdate({ ...workflow, status: change.status })) {
  throw new HttpException(
    "The selected assignee's role is not sufficient for the new status.",
    HttpStatus.UNPROCESSABLE_ENTITY,
  );
}
```